KafkaIngester/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -66,32 +256,251 @@ public static Table simpleConsumeToTable(
+            @NotNull final String typeProperty,
+            @NotNull final String deserializerProperty,
+            @NotNull final String nameProperty,
+            @NotNull final String nameDefault,
+            final boolean allowEmpty) {
+        final String columnName;
+        if (properties.containsKey(nameProperty)) {
+            columnName = properties.getProperty(nameProperty);
+            if (columnName == null || columnName.equals("")) {
+                if (allowEmpty) {
+                    return null;
+                }
+                throw new IllegalArgumentException("Property for " + nameDefault + " can't be empty.");
+            }
+        } else {
+            columnName = nameDefault;
+        }
+
+        if (properties.containsKey(typeProperty)) {
+            final String typeAsString = properties.getProperty(typeProperty);
+            switch (typeAsString) {
+                case "int":
+                    properties.setProperty(deserializerProperty, INT_DESERIALIZER);
+                    return ColumnDefinition.ofInt(columnName);
+                case "long":
+                    properties.setProperty(deserializerProperty, LONG_DESERIALIZER);
+                    return ColumnDefinition.ofLong(columnName);
+                case "double":
+                    properties.setProperty(deserializerProperty, DOUBLE_DESERIALIZER);
+                    return ColumnDefinition.ofDouble(columnName);
+                case "byte[]":
+                    properties.setProperty(deserializerProperty, BYTE_ARRAY_DESERIALIZER);
+                    return ColumnDefinition.fromGenericType(columnName, byte[].class, byte.class);
+                case "String":
+                case "string":
+                    properties.setProperty(deserializerProperty, STRING_DESERIALIZER);
+                    return ColumnDefinition.ofString(columnName);
+                default:
+                    throw new IllegalArgumentException(
+                            "Property " + typeProperty + " value " + typeAsString + " not supported");
+            }
+        } else if (!properties.containsKey(deserializerProperty)) {
+            properties.setProperty(deserializerProperty, STRING_DESERIALIZER);
+            return ColumnDefinition.ofString(columnName);
+        }
+        final String deserializer = properties.getProperty(deserializerProperty);
+        if (INT_DESERIALIZER.equals(deserializer)) {
+            return ColumnDefinition.ofInt(columnName);
+        }
+        if (LONG_DESERIALIZER.equals(deserializer)) {
+            return ColumnDefinition.ofLong(columnName);
+        }
+        if (DOUBLE_DESERIALIZER.equals(deserializer)) {
+            return ColumnDefinition.ofDouble(columnName);
+        }
+        if (BYTE_ARRAY_DESERIALIZER.equals(deserializer)) {
+            return ColumnDefinition.fromGenericType(columnName, byte[].class, byte.class);
+        }
+        throw new IllegalArgumentException(
```

Comment:
why no string here?

Reply:
I was thinking about it wrong.  We need one.